aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -114,6 +114,95 @@ Tensor flip_mps(const Tensor& self, IntArrayRef dims) {
 
 }
 
+TORCH_IMPL_FUNC(index_add_mps_out)(
+  const Tensor& self,
+  int64_t dim,
+  const Tensor& index,
+  const Tensor& source,
+  const Scalar& alpha,
+  const Tensor& result) {
+  auto numel = index.numel();
+
+  using namespace mps;
+  MPSStream* stream = getCurrentMPSStream();
+  dim = maybe_wrap_dim(dim, self.dim());
+
+  if (numel == 0) {
+    return;
+  }
+
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor* inputTensor_ = nil;
+    MPSGraphTensor* indexTensor_ = nil;
+    MPSGraphTensor* sourceTensor_ = nil;
+    MPSGraphTensor* outputTensor_ = nil;
+  };
+
+  MPSGraphCache* cache_ = MPSGraphCache::getInstance();
+
+  @autoreleasepool {
+
+    string key = "index_add_mps_out" + getTensorsStringKey({self, index, source}) + ":" + std::to_string(dim) + ":" + std::to_string(alpha.to<double>());
```

Comment:
`alpha` being in the key can be a reason for concern, if it keeps changing. It will result in us creating graphs. Can you create a RankedPlaceholder for the `alpha` by following the same convention:
```
MPSGraphTensor* alphaTensor = mpsGraphRankedPlaceHolder(mpsGraph, alpha);

// and then passing in feeds the value
NSDictionary<MPSGraphTensor*, MPSGraphTensorData*>* feeds = @{
```

Reply:
Done.